src/methods/properties.js

```diff
@@ -1005,7 +1005,16 @@ export class PropertiesImplementation {
         // ii. Set base to ToObject(base).
         base = To.ToObjectPartial(realm, base);
       }
-      invariant(base instanceof ObjectValue || base instanceof AbstractObjectValue);
+      if (!(base instanceof ObjectValue || base instanceof AbstractObjectValue)) {
+        let diagnostic = new CompilerDiagnostic(
+          "cannot PutValue on a base value that is not an object or abstract object",
```

Comment:
The reader of this message is thinking about JavaScript source language constructs and may not find a reference to "PutValue" to be very informative.